worker/src/lib/HeliconeProxyRequest/forwarder.ts

```diff
@@ -130,44 +173,98 @@ export async function proxyForwarder(
     }
   }
 
-  // TODO move this to proxyRequest
-  const { data: cacheSettings, error: cacheError } = getCacheSettings(
-    proxyRequest.requestWrapper.getHeaders(),
-    proxyRequest.isStream
-  );
+  if (
+    proxyRequest.requestWrapper.heliconeHeaders.moderationsEnabled &&
+    provider == "OPENAI"
+  ) {
+    let latestMessage;
 
-  if (cacheError !== null) {
-    return responseBuilder.build({
-      body: cacheError,
-      status: 500,
-    });
-  }
+    try {
+      latestMessage = JSON.parse(proxyRequest.bodyText ?? "").messages.pop();
+    } catch (error) {
+      console.error("Error parsing latest message:", error);
+      return responseBuilder.build({
+        body: "Failed to parse the latest message.",
+        status: 500,
+      });
+    }
 
-  if (cacheSettings.shouldReadFromCache) {
-    const { data: auth, error: authError } = await request.auth();
-    if (authError == null) {
-      const db = new DBWrapper(env, auth);
-      const { data: orgData, error: orgError } = await db.getAuthParams();
-      if (orgError !== null || !orgData?.organizationId) {
-        console.error("Error getting org", orgError);
-      } else {
-        const cachedResponse = await getCachedResponse(
-          proxyRequest,
-          cacheSettings.bucketSettings,
-          env.CACHE_KV,
-          cacheSettings.cacheSeed
-        );
-        if (cachedResponse) {
-          ctx.waitUntil(
-            recordCacheHit(
-              cachedResponse.headers,
-              env,
-              new ClickhouseClientWrapper(env),
-              orgData.organizationId
-            )
-          );
-          return cachedResponse;
+    if (latestMessage.role == "user") {
+      const moderationRequest = new Request(
+        "https://api.openai.com/v1/moderations",
+        {
+          method: "POST",
+          headers: proxyRequest.requestWrapper.headers,
+          body: JSON.stringify({
+            input: latestMessage.content,
```

Comment:
content might not exist on lastMessage

Reply:
Should we return an error when the content is empty/does not exist?

Reply:
IMO, if no content, we just don't check moderation and continue. Maybe there is a reason and we want the provider to fail, not us. 

I would add to the latestMessage?.role if check if the content exists